### ISSUE TITLE
Writing operators misses a space

### ISSUE THREAD

**Issue by UWN** [ichiban/prolog#318]
```
?- writeq(1 is X),nl.
1 is_112
X = _112.
```
Expected: `1 is _112`

See [7.10.5 h 1](https://www.complang.tuwien.ac.at/ulrich/iso-prolog/draft-7.10.5#h_1):

> ... In all cases, a space is output to separate an operator from its argument(s) if any ambiguity could otherwise arise.

Similarly [variable_names#73](https://www.complang.tuwien.ac.at/ulrich/iso-prolog/variable_names#73).

**Comment by ichiban** [comment]
Thank you very much for reporting! Fixed in v1.2.1.

```console
$ go install github.com/ichiban/prolog/cmd/1pl@latest
go: downloading github.com/ichiban/prolog v1.2.1
$ $(go env GOPATH)/bin/1pl
Top level for ichiban/prolog v1.2.1
This is for testing purposes only!
See https://github.com/ichiban/prolog for more details.
Type Ctrl-C or 'halt.' to exit.
?- writeq(1 is X),nl.
1 is _79
X = _79.
?- S=(1 is T),N='X',write_term(S,[quoted(true), variable_names([N=T])]),nl.
1 is X
N = 'X',
S = 1 is _90,
T = _90.
?- 
```

**Comment by UWN** [comment]
Back to [perfect scores](https://www.complang.tuwien.ac.at/ulrich/iso-prolog/variable_names#73)!
